Sources/Testing/Running/Runner.swift

```diff
@@ -114,6 +114,54 @@ extension Runner {
     }
   }
 
+  /// Execute the ``CustomExecutionTrait/execute(_:for:)`` functions associated
+  /// with the test in a plan step.
+  ///
+  /// - Parameters:
+  ///   - step: The step being performed. If this argument is `nil`, `body` is
+  ///     immediately executed.
+  ///   - body: A function to execute from within the
+  ///           ``CustomExecutionTrait/execute(_:for:)`` functions of each trait
+  ///            applied to `step.test`.
+  ///
+  /// - Throws: Whatever is thrown by `body` or by any of the
+  ///   ``CustomExecutionTrait/execute(_:for:)`` functions.
+  ///
+  /// If `step` is not `nil`, this function also sets the current test (via
+  /// ``Test/current``) during the execution of `body`. This allows traits'
+  /// ``CustomExecutionTrait/execute(_:for:)`` functions to record issues, etc.
+  private func _executeTraits(
+    for step: Plan.Step?,
+    _ body: @escaping () async throws -> Void
+  ) async throws {
+    // If this step is nil, simply execute `body` as-is.
+    guard let test = step?.test else {
+      return try await body()
+    }
+
+    if case .skip = step?.action {
+      return try await body()
+    }
+
+    // Construct a recursive function that invokes each trait's ``execute(_:for:)``
+    // function. The order of the sequence is reversed so that the last trait is
```

Comment:
I'm curious about the behavior of **reversing** the traits here and whether that's the right decision. I think it will also mean that traits applied to containing suite types and inherited would also have their execute methods invoked _after_ traits directly applied to a test function, and conceptually that feels a bit wrong to me. Does anyone have usage examples in mind for the ordering?

(Either way, we should document what we decide here in the DocC docs for this protocol method.)

Reply:
Reversing the order means we invoke them in the order they're specified. It's counter-intuitive but it's a side effect of the nested calls. See the comment here.